tachyon/zk/plonk/constraint_system.h

```diff
@@ -247,7 +247,7 @@ class ConstraintSystem {
   // panic if |constrain| returns an empty iterator.
   void CreateGate(std::string_view name, ConstrainCallback constrain) {
     VirtualCells cells(this);
-    std::vector<Constraint<F>> constraints = std::move(constraints).Run(cells);
+    std::vector<Constraint<F>> constraints = std::move(constrain).Run(cells);
```

Comment:
```suggestion
    std::vector<Constraint<F>> constraints = std::move(constraint).Run(cells);
```

Reply:
Is `constrain` a typo or intended? I think using a noun is more suitable than a verb in this case.

Reply:
I name it with a verb on purpose like `assign` for `AssignaCallback`